web/files/files.go

```diff
@@ -208,46 +218,61 @@ func ModificationHandler(c *gin.Context) {
 	}
 
 	patchData := container.Data
-	docType, err := vfs.ParseDocType(patchData.Type)
+
+	fileID := c.Param("file-id")
+
+	var docType vfs.DocType
+	if fileID == MetadataPath {
+		docType, err = vfs.ParseDocType(c.Query("Type"))
+	} else {
+		docType, err = vfs.ParseDocType(patchData.Type)
+	}
+
 	if err != nil {
 		jsonapi.AbortWithError(c, WrapVfsError(err))
 		return
 	}
 
-	var doc jsonapi.Object
-	switch docType {
-	case vfs.FileDocType:
-		doc, err = modFileHandler(vfsC, c.Request, patchData)
-	case vfs.FolderDocType:
-		// @TODO
-		err = fmt.Errorf("Not implemented")
+	var doc couchdb.Doc
+	if fileID == MetadataPath {
+		switch docType {
+		case vfs.FileDocType:
+			doc, err = vfs.GetFileDocFromPath(vfsC, c.Query("path"))
+		case vfs.FolderDocType:
+			doc, err = vfs.GetDirectoryDocFromPath(vfsC, c.Query("path"))
+		}
```

Comment:
This error is handled by the `ParseDocType` function. But I can be more explicit by having another check.